### PR TITLE
CIRCSTORE-400 Implement search fields migration

### PR DESCRIPTION
**Pull request by alexanderkurash**
Resolves [CIRCSTORE-400](https://issues.folio.org/browse/CIRCSTORE-400)

## Purpose
Migrate fields needed for request search index

## Approach
Create a service that would process all requests (in batches), fetch service points and items and update search index fields

### CODE REVIEW

File: src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    batch.getRequestMigrationContexts().forEach(ctx -> {
+      ctx.setCallNumberComponents(itemIdToCallNumberComponents.get(ctx.getItemId()));
+      ctx.setShelvingOrder(itemIdToShelvingOrder.get(ctx.getItemId()));
+    });
+  }
+
+  public void buildNewRequest(RequestSearchMigrationContext context) {
+    final JsonObject migratedRequest = context.getOldRequest().copy();
+
+    write(migratedRequest, "pickupServicePointName", context.getPickupServicePointName());
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java:123]: shouldn't these fields go to "searchIndex"?

Comment (alexanderkurash) [reply src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java]: Fixed

File: src/main/java/org/folio/service/migration/AbstractRequestMigrationService.java [hunk 2]
```diff
@@ -168,9 +170,11 @@ public Future<Batch> validateRequests(Batch batch) {
       : failedFuture(join(lineSeparator(), errors));
   }
 
-  abstract Future<Void> processBatch(Batch batch);
+  abstract Future<Void> processBatch(Batch<T> batch);
 
-  abstract Collection<String> validateRequest(RequestMigrationContext context);
+  Collection<String> validateRequest(T context) {
+    return List.of();
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/migration/AbstractRequestMigrationService.java:176]: why empty list?

File: src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    log.info("findCallNumbers:: batch={}", batch);
+
+    Set<String> itemIds = batch.getRequestMigrationContexts()
+      .stream()
+      .map(RequestSearchMigrationContext::getItemId)
+      .filter(Objects::nonNull)
+      .collect(toSet());
+
+    if (itemIds.isEmpty()) {
+      return succeededFuture(batch);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java:106]: add logging like for service points above?

File: src/test/java/org/folio/rest/api/TenantRefApiTests.java [hunk 4]
```diff
@@ -161,69 +168,133 @@ public static void afterAll(final TestContext context) throws Exception {
...
       .onSuccess(job -> {
         context.assertTrue(job.getError().contains("Request failed: GET"));
         context.assertTrue(job.getError().contains("Response: [404]"));
-        assertThatNoRequestsWereUpdated(context);
+        assertThatNoRequestsWereUpdatedByTlrMigration(context);
         async.complete();
       });
   }
 
+  ////
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/api/TenantRefApiTests.java:272]: ???

File: src/test/java/org/folio/rest/api/TenantRefApiTests.java [hunk 5]
```diff
@@ -161,69 +168,133 @@ public static void afterAll(final TestContext context) throws Exception {
...
+  private void requestSearchMigrationFailsWhenRemoteCallFails(TestContext context, Async async) {
+    postTenant(context, REQ_SEARCH_MIGRATION_PREV_MOD_VER, REQ_SEARCH_MIGRATION_MOD_VER)
+      .onSuccess(job -> {
+        context.assertTrue(job.getError().contains("Request failed: GET"));
+        context.assertTrue(job.getError().contains("Response: [404]"));
+        assertThatNoRequestsWereUpdatedByRequestSearchMigration(context);
+        async.complete();
+      });
+  }
+  ///
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/api/TenantRefApiTests.java:296]: ???

File: src/test/java/org/folio/rest/api/TenantRefApiTests.java (hunk 4 above)
Comment (alexanderkurash) [reply src/test/java/org/folio/rest/api/TenantRefApiTests.java:272]: removed

File: src/test/java/org/folio/rest/api/TenantRefApiTests.java (hunk 5 above)
Comment (alexanderkurash) [reply src/test/java/org/folio/rest/api/TenantRefApiTests.java:296]: removed

File: src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java (hunk 3 above)
Comment (alexanderkurash) [reply src/main/java/org/folio/service/migration/RequestSearchFieldsMigrationService.java:106]: added

**Pull request merged**